drivers/lte_link_control/lte_lc.c

```diff
@@ -423,6 +436,338 @@ int lte_lc_edrx_req(bool enable)
 	return 0;
 }
 
+/**@brief Helper function to check if a response is what was expected
+ *
+ * @param response Pointer to response prefix
+ * @param response_len Length of the response to be checked
+ * @param check The buffer with "truth" to verify the response against,
+ *		for example "+CEREG"
+ *
+ * @return True if the provided buffer and check are equal, false otherwise.
+ */
+static bool response_is_valid(const char *response, size_t response_len,
+			      const char *check)
+{
+	if ((response_len < strlen(check)) ||
+	    (memcmp(response, check, response_len) != 0)) {
+		return false;
+	}
+
+	return true;
+}
+
+/**@brief Parses an AT command response, and returns the current network
+ *	  registration status if it's available in the string.
+ *
+ * @param at_response Pointer to buffer with AT response.
+ * @param status Pointer to where the registration status is stored.
+ *
+ * @return Zero on success or (negative) error code otherwise.
+ */
+static int parse_nw_reg_status(const char *at_response,
+			       enum lte_lc_nw_reg_status *status)
```

Comment:
"status" needs to be cheked against NULL - "at_response" is checked by at_parser_max_params_from_str()

Reply:
Fixed (added check for both, though a tiny bit redundant)